lib/govuk_schemas/random_example.rb

```diff
@@ -81,15 +81,25 @@ def self.for_schema(schema_key_value, &block)
     # @raise [GovukSchemas::InvalidContentGenerated]
     def payload
       payload = @random_generator.payload
-      # ensure the base payload is valid
-      errors = validation_errors_for(payload)
-      raise InvalidContentGenerated, error_message(payload, errors) if errors.any?
 
       if block_given?
+        original_payload = payload
         payload = yield(payload)
-        # check the payload again after customisation
+
+        customised_errors = validation_errors_for(payload)
+
+        if customised_errors.any?
+          # check with the original payload had errors to produce the correct message
+          original_errors = validation_errors_for(original_payload)
+          errors = original_errors.any? ? original_errors : customised_errors
+          error_payload = original_errors.any? ? original_payload : payload
+          message = error_message(error_payload, errors, customised: original_errors.empty?)
+
+          raise InvalidContentGenerated, message
+        end
```

Comment:
There's not actually a change in the behaviour from how it was previously. In the first case you referenced the code would previously error about the original item: https://github.com/alphagov/govuk_schemas/blob/0cb2983c38770075ed80a18f6973cd1bad70a923/lib/govuk_schemas/random_example.rb#L83-L96

So the change in this is just the way it reaches outputting those errors. Previously it would validate the original item then customise it then validate the customised item (2 rounds of validation). Now it would validate the customised item and only if that failed validate the original item (1 round of validation unless there is a failure).

Only difference in behaviour for a user is an edge case scenario where a random original payload that is invalid is made valid by customisation. Previously that would error but now it will not. However I don't think anyone would find that behaviour problematic and need to know about it.